### PR TITLE
Faketensor hpu device normalization

### PR DESCRIPTION
**Pull request by VRSinghHabana**
FakeTensor doesn't normalize device_idx  and failed with below testcase.

import torch
import habana_frameworks.torch.hpu
from torch._subclasses.fake_tensor import FakeTensorMode

with FakeTensorMode.push():
    a = torch.empty(1, device="hpu")
    b = torch.empty(1, device="hpu:0")
    result = a + b

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Not sure if we should normalize all devices this way?
In any case that sounds ok

FYI @ezyang
